Assets/HoloToolkit/Utilities/Tests/Scripts/PlaneTargetGroupPicker.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using System.Collections;
+using HoloToolkit.Unity;
+
+public class PlaneTargetGroupPicker : Singleton<PlaneTargetGroupPicker>
+{
+    [Tooltip("In degrees")]
+    public float AngleOfAcceptance = 45.0f;
+    public PlaneTargetGroup[] Groups;
+
+    public TextMesh DisplayText;
+    public float TextDisplayTime = 5.0f;
+
+    private PlaneTargetGroup currentGroup;
+
+    public void PickNewTarget()
+    {
+        PlaneTargetGroup newGroup = null;
+        float smallestAngle = float.PositiveInfinity;
+
+        // Figure out which group we're looking at
+        foreach (PlaneTargetGroup group in Groups)
+        {
+            Vector3 camToGroup = group.transform.position - Camera.main.transform.position;
+            float gazeObjectAngle = Vector3.Angle(camToGroup, Camera.main.transform.forward);
+            if (group.Targets.Length > 0 && gazeObjectAngle < AngleOfAcceptance && gazeObjectAngle < smallestAngle)
+            {
+                smallestAngle = gazeObjectAngle;
+                newGroup = group;
+            }
+        }
+
+        // Looking at a group!
+        if (newGroup != null)
+        {
+            // If we're already in this group, switch targets
+            if (newGroup == currentGroup)
+            {
+                newGroup.PickNewTarget();
+            }
+            currentGroup = newGroup;
+            StabilizationPlaneModifier.Instance.TargetOverride = currentGroup.CurrentTarget.transform;
+            UpdateText();
+        }
+    }
+
+    private void UpdateText()
```

Comment:
If you `PickNewTarget` faster than `TextDisplayTime` you'll get overlapping set/clears of the text because you'll spawn multiple coroutines of DisplayForSeconds and updated text won't last the full `TextDisplayTime`.